### PR TITLE
[AAE-10644] Fix tests that have no expectations - core lib part 2

### PR DESCRIPTION
**Pull request by tomgny**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://alfresco.atlassian.net/browse/AAE-10644

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/src/lib/directives/node-delete.directive.spec.ts [hunk 1]
```diff
@@ -271,16 +271,16 @@ describe('NodeDeleteDirective', () => {
             });
         });
 
-        it('should emit event when delete is done', (done) => {
+        it('should emit event when delete is done', async () => {
```
Comment (DenysVuika) [inline lib/core/src/lib/directives/node-delete.directive.spec.ts:274]: you make it async but not using await in the code, I suggest converting observable to promise and awaiting it

File: lib/core/src/lib/form/components/widgets/core/form.model.spec.ts [hunk 2]
```diff
@@ -284,18 +284,22 @@ describe('FormModel', () => {
         expect(form.outcomes[1].isSystem).toBeFalsy();
     });
 
-    it('should raise validation event when validating form', (done) => {
+    it('should raise validation event when validating form', () => {
         const form = new FormModel({}, null, false, formService);
 
-        formService.validateForm.subscribe(() => done());
+        formService.validateForm.subscribe((validateFormEvent) =>
+            expect(validateFormEvent).toBeTruthy()
```
Comment (DenysVuika) [inline lib/core/src/lib/form/components/widgets/core/form.model.spec.ts:291]: this expect may not wait for the test end

File: lib/core/src/lib/form/components/widgets/core/form.model.spec.ts [hunk 3]
```diff
@@ -284,18 +284,22 @@ describe('FormModel', () => {
...
     });
 
-    it('should raise validation event when validating field', (done) => {
+    it('should raise validation event when validating field', () => {
         const form = new FormModel({}, null, false, formService);
         const field = jasmine.createSpyObj('FormFieldModel', ['validate']);
 
-        formService.validateFormField.subscribe(() => done());
+        formService.validateFormField.subscribe((validateFormFieldEvent) =>
+            expect(validateFormFieldEvent).toBeTruthy()
```
Comment (DenysVuika) [inline lib/core/src/lib/form/components/widgets/core/form.model.spec.ts:301]: this expect may not wait for the test end

File: lib/core/src/lib/login/components/login.component.spec.ts [hunk 4]
```diff
@@ -173,7 +173,7 @@ describe('LoginComponent', () => {
         expect(router.navigateByUrl).toHaveBeenCalledWith('some-route');
     }));
 
-    it('should update user preferences upon login', fakeAsync(() => {
+    it('should update user preferences upon login', async () => {
```
Comment (DenysVuika) [inline lib/core/src/lib/login/components/login.component.spec.ts:176]: await not used in the test

File: lib/core/src/lib/services/people-process.service.spec.ts [hunk 5]
```diff
@@ -65,7 +67,7 @@ describe('PeopleProcessService', () => {
             jasmine.Ajax.uninstall();
         });
 
-        it('should be able to retrieve people to involve in the task', (done) => {
+        it('should be able to retrieve people to involve in the task', async () => {
```
Comment (DenysVuika) [inline lib/core/src/lib/services/people-process.service.spec.ts:70]: it's hard to follow the pattern, you change "done" to async in one cases, and then change to "fakeAsync" in other

Comment (DenysVuika) [reply lib/core/src/lib/services/people-process.service.spec.ts:70]: why it's async/await here, and next tests with similar cases are "fakeAsync" ?

**Review by DenysVuika — CHANGES_REQUESTED** [review]

File: lib/core/src/lib/form/components/widgets/core/form.model.spec.ts (hunk 2 above)
Comment (tomgny) [reply lib/core/src/lib/form/components/widgets/core/form.model.spec.ts:291]: this test is synchronous, so we don't need to wait for anything here, we just checking if the validation event was raised. before we were using done method to check if it will reach subscription, instead of that we can expect some validation event.

File: lib/core/src/lib/form/components/widgets/core/form.model.spec.ts (hunk 3 above)
Comment (tomgny) [reply lib/core/src/lib/form/components/widgets/core/form.model.spec.ts:301]: the same as comment above

File: lib/core/src/lib/services/people-process.service.spec.ts (hunk 5 above)
Comment (tomgny) [reply lib/core/src/lib/services/people-process.service.spec.ts]: it was done by mistake, fixed

File: lib/core/src/lib/login/components/login.component.spec.ts (hunk 4 above)
Comment (tomgny) [reply lib/core/src/lib/login/components/login.component.spec.ts:174]: await added

File: lib/core/src/lib/directives/node-delete.directive.spec.ts (hunk 1 above)
Comment (tomgny) [reply lib/core/src/lib/directives/node-delete.directive.spec.ts:260]: await added
